### PR TITLE
feat(comp): Radio component rewrite

### LINKED ISSUE #1570: <Radio> has no prefix for its variant and hardcoded variants
## Description

Radio is missing a prefix for its ' variant'. We should add a prefix and also add it to the `Theme`.
Radio shouldn't have hardcoded variants.

## Context

We don't want to force the Theme to have specific variants in a component.
Radio should have a variant.

## Consequences

Easier to write and access a theme. Easier to use the Radio component.

### PR DESCRIPTION
**Pull request by ti10le**
closes #1570

### CODE REVIEW

File: packages/components/src/Radio/Radio.tsx [hunk 1]
```diff
@@ -101,28 +63,26 @@ export const Radio: React.FC<RadioProps> = ({
...
-      </>
-    );
-  }
-
-  return <RadioInput {...props} />;
-};
+}) => (
+  <>
+    <Box
+      as={Label}
```
Comment (sebald) [inline packages/components/src/Radio/Radio.tsx:69]: Why no `<Label>` ?

File: packages/components/src/Radio/Radio.tsx [hunk 2]
```diff
@@ -16,78 +17,39 @@ export interface RadioThemeExtension<Value> {
...
   return (
-    <Box
-      as={RadioUnchecked}
-      variant={`radio.${variant}`}
-      disabled={disabled}
-      error={error}
-    />
+    <Box pr="xsmall">
+      <VisuallyHidden>
+        <Box
```
Comment (sebald) [inline packages/components/src/Radio/Radio.tsx:30]: No need to use the `<Box>` here we don't style it anyway

File: packages/components/src/Radio/RadioIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    >
+      <Box
+        variant={conditional(`radio.${variant}`, conditionalStates)}
+        as="circle"
+        cx="8"
+        cy="16"
+        r="7.5"
+      />
+      {checked && (
+        <Box __baseCSS={{ fill: 'gray00' }} as="circle" cx="8" cy="16" r="3" />
```
Comment (sebald) [inline packages/components/src/Radio/RadioIcon.tsx:47]: No need to use `<Box>` you can use a regular cicle here.

File: .changeset/sharp-shrimps-bake.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
```
Comment (sebald) [inline .changeset/sharp-shrimps-bake.md:1]: One changeset is enough? :D

File: packages/components/src/Radio/Radio.test.tsx [hunk 5]
```diff
@@ -98,22 +103,29 @@ test('supports error and errorMessage prop', () => {
...
+        disabled
+      />
     </ThemeProvider>
   );
 
   const radio = screen.getByTitle(/radio/);
-  expect(radio).toBeDefined();
+  expect(radio).toHaveAttribute('checked');
   expect(radio).toHaveAttribute('disabled');
 });
```
Comment (sebald) [inline packages/components/src/Radio/Radio.test.tsx:131]: Add an interaction test!? :)

**Review by sebald — CHANGES_REQUESTED** [review]

**Review by sebald — CHANGES_REQUESTED** [review]
see comments

File: packages/components/src/Radio/Radio.tsx (hunk 1 above)
Comment (ti10le) [reply packages/components/src/Radio/Radio.tsx:69]: because of the __baseCSS prop

File: packages/components/src/Radio/RadioIcon.tsx (hunk 3 above)
Comment (ti10le) [reply packages/components/src/Radio/RadioIcon.tsx:47]: but how to use the fill prop? Is it better to use `fill='white'` ?

File: packages/components/src/Radio/Radio.tsx (hunk 1 above)
Comment (sebald) [reply packages/components/src/Radio/Radio.tsx:68]: I don't see where the difference between a hover effect and the color is. Setting the color like you did will override anything. Why not using `css` prop and do this with the cursor too? :)

File: packages/components/src/Radio/RadioIcon.tsx (hunk 3 above)
Comment (sebald) [reply packages/components/src/Radio/RadioIcon.tsx:47]: `gray80` isn't part of any theme too and might me just undefined :-/

Comment (ti10le) [reply packages/components/src/Radio/RadioIcon.tsx:47]: Means this "yes" to my question?

File: packages/components/src/Radio/Radio.tsx (hunk 1 above)
Comment (ti10le) [reply packages/components/src/Radio/Radio.tsx:68]: Yes, I can use cursor and color in the css prop. Then we wouldn't need the color prop on the Label.
At the end we have also Box as label.
```
 <Box
      as={Label}
      htmlFor={props.id}
      required={required}
      variant={`label.${labelVariant}`}
      css={
        props.disabled
          ? { color: 'disabled', ':hover': { cursor: 'not-allowed' } }
          : { color: 'text', ':hover': { cursor: 'pointer' } }
      }
    >
```

Comment (sebald) [reply packages/components/src/Radio/Radio.tsx:68]: Ah right. No CSS prop on the label. My bad, sorry.

Makes me think of we should expose the css prop on more components actually 😄 but this is another issue

**Review by sebald — APPROVED** [review]

**Pull request merged**